docs/common/istio-ambient-mode.md

```diff
@@ -0,0 +1,366 @@
+
+## Introduction to Istio Ambient mode
+
+Ambient mesh was announced in September 2022 and reached a stable state in the Istio 1.24 release. It's core innovation is the separation of Layer 4 (L4) and Layer 7 (L7) processing into two distinct layers. It uses lightweight, shared L4 node proxies and optional L7 proxies, eliminating the need for traditional sidecar proxies in the data plane.
+
+The lightweight, shared L4 node proxy is called ztunnel (zero-trust tunnel). It significantly reduces the overhead of running a mesh by eliminating the need to over-provision memory and CPU while still ensuring security through mutual TLS, simple L4 authorization policies, and telemetry.
+
+The L7 proxies, known as waypoints, handle functions like traffic routing, authorization policy enforcement, and resilience. They run separately from application deployments and can scale independently based on demand.
+
+In contrast to the Istio Sidecar mode. Users can start with the secure L4 overlay, which offers features such as mTLS, authorization policy, and telemetry. Complex L7 handling such as retries, traffic splitting, load balancing, and observability collection can then be enabled on a case-by-case basis.
```

Comment:
Hi @yxun I still don't understand this sentence. "In contrast to the Istio Sidecar mode. Users can start with the secure L4 overlay..."

Reply:
Hello @nyechiel, @yxun is on PTO for the rest of the week, so let me try if I can clarify it. 

The main difference from Istio’s Sidecar mode is that with Ambient mode, we don’t have to enable full Layer 7 features right away. We can start with a lightweight, secure Layer 4 overlay, which gives us features like mTLS, authorization policies, and telemetry. If we need more advanced L7 features, like retries, traffic splitting etc, we can create waypoints as needed. In Sidecar mode, L7 processing is always included by default, so we don’t have this flexibility.

Please feel free to suggest/comment on any improvements. Thank you.

Reply:
Oh yeah, good catch! There should be a "," instead of "."

Reply:
I will include this change in another PR when I finish a waypoint user guide draft early this week.

Reply:
this issue is addressed in PR https://github.com/istio-ecosystem/sail-operator/pull/694